src/v/rpc/transport.cc

```diff
@@ -200,50 +200,61 @@ transport::do_send(sequence_t seq, netbuf b, rpc::client_opts opts) {
                   seq, std::make_unique<entry>(std::move(e)));
                 dispatch_send();
                 return std::move(f).finally([u = std::move(units)] {});
-            })
-            .finally([this, seq] {
-                // update last sequence to make progress, for successfull
-                // dispatches this will be noop, as _last_seq was already update
-                // before sending data
-                _last_seq = std::max(_last_seq, seq);
             });
       });
 }
 
 void transport::dispatch_send() {
+    if (_is_sending_background) {
+        _has_skipped_dispatch = true;
+        return;
+    }
+
+    _is_sending_background = true;
```

Comment:
> rpc/transport: ensure only one dispatch_send bg task is running.
If multiple `dispatch_send` calls are made concurrently it is
possible for the `ordering_test` in `rpc_gen_cycling_test.cc` to
fail.

tbh the rpc code is quite fragile and i'm hesitant to change it without a thorough understanding of what is happening. can you provide an analysis of what is wrong with the transport, why the test is failing, and what the fix does?

Reply:
+1, this commit is the riskiest in the whole PR.  Need to understand very clearly what is happening + why it's correct.